YellowNotes.Api/Filters/ValidateHttpContextClaimsFilter.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using YellowNotes.Api.Extensions;
+
+namespace YellowNotes.Api.Filters
+{
+    public class ValidateHttpContextClaimsFilter : ActionFilterAttribute
+    {
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            var userEmail = context.HttpContext.GetEmailFromClaims();
+            if (userEmail != null)
+            {
+                return;
+            }
+
+            context.Result = new BadRequestObjectResult(
```

Comment:
It can't be BadRequest. when claims don't have email it means that user is not authenticated.